lib/enum/value.rb

```diff
@@ -0,0 +1,143 @@
+module Enum
+  class Value
+    
+    # TODO: Perhaps stored_value should be renamed unsafe_value?
+    
+    include Comparable
+
+    attr_reader :stored_value, :error
+    
+    class << self
+      attr_accessor :default_value, :suppress_read_errors, :klass
+    end
+    
+    def self.inherited(subclass)
+      # Value subclass settings and options
+      #   @default_value => <:ERROR|:ANY|:something|nil>
+      #   @suppress_read_errors => <true|false>
+      subclass.suppress_read_errors = false
+      subclass.default_value = :ERROR
```

Comment:
It looks strange that default value is `ERROR`, why not `ANY`? 🤔 
Also I don't like capitalized versions... May be `_error`, `$error` and etc. It needs to be elaborated more. We need to come up with a name that potentially won't conflict with user input.

Reply:
On the spectrum between safe and convenient, I went with safe, mostly because the title of the gem is `safe-enum` :smile: . The default behavior will raise an exception whenever the Value object is loaded or read with an invalid token. There are goods and bads about either of the 'special' default settings.

1. With a safe default `:ERROR`, users are more likely to raise exceptions that stem from data issues, especially if using this to model data pulled from a database.

2. With a relaxed default `:ANY`, users will get value objects without exceptions, though the objects may be invalid. If this is the default for `default_value`, should `suppress_read_errors` also default to `true`?

A possible middle ground is a default value of nil. The user won't accidentally use an invalid value, yet they won't be interrupted by exceptions when loading data from a db. They can still check for `valid?` and look at `@error` to get more information.

Your suggestions for alternatives to the special defaults are fine. `:$error` is interesting, I haven't considered  `$` in a symbol name before. Probably much less chance of clashing with user input.

Reply:
Ok, it's clear. But I'm still for renaming capitalized words to lowercase (I mean `:$error`).

Reply:
No problem, I'll rename those per your suggestions.